reference/conanfile/methods/requirements.rst

```diff
@@ -0,0 +1,121 @@
+.. _reference_conanfile_methods_requirements:
+
+requirements()
+==============
+
+Requirement traits
+^^^^^^^^^^^^^^^^^^
+
+Traits are properties of a requires clause. They determine how various parts of a
+dependency are treated and propagated by Conan. Values for traits are usually computed by
+Conan based on dependency's :ref:`reference_conanfile_attributes_package_type`, but can
+also be specified manually.
+
+A good introduction to traits is provided in the `Advanced Dependencies Model in Conan 2.0
```

Comment:
asking to watch 1 hour long video in documentation is kinda inconvenient.
is it possible to extract relevant parts of the material here instead?

Reply:
This section I didn't write it, it was there, I just moved it.
This will be done later in other PRs, for this reference of methods, this is good enough atm